Make running c5 on a non-standard port work

Mnkras opened the pull request:
Fixes #2069

Added this like and ports showed up everywhere so I guess it works, definitely works better than it did before :P

mlocati commented:
This does not seem to work... I just fetched the latest code from GitHub, applied this patch, re-installed via my batch script, but I still have links without my custom port...

![no-custom-port](https://cloud.githubusercontent.com/assets/928116/6683168/91514548-cc82-11e4-8075-dc373e6c10fc.png)

Mnkras commented:
I had to clear c5s cache manually before it worked, it seems to be
aggressively cached.

On Tue, Mar 17, 2015, 3:50 AM Michele Locati <notifications@github.com>
wrote:

> This does not seem to work... I just fetched the latest code from GitHub,
> applied this patch, re-installed via my batch script, but I still have
> links without my custom port...
>
> [image: no-custom-port]
> <https://cloud.githubusercontent.com/assets/928116/6683168/91514548-cc82-11e4-8075-dc373e6c10fc.png>
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/concrete5/concrete5-5.7.0/pull/2074#issuecomment-82189256>
> .
>

mlocati commented:
My install script completly deletes the /application/files folder, so even the most aggressive caching system in the universe shouldn't survive :wink:

Mnkras commented:
Weird, it's working for me..

On Tue, Mar 17, 2015, 10:26 AM Michele Locati <notifications@github.com>
wrote:

> My install script completly deletes the /application/files folder, so even
> the most aggressive caching system in the universe shouldn't survive [image:
> :wink:]
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/concrete5/concrete5-5.7.0/pull/2074#issuecomment-82382118>
> .
>

mlocati commented:
I did a fast check with a step-by-step debug, and the function that gets called is [\Concrete\Core\Url\Url::createFromUrl()](https://github.com/concrete5/concrete5-5.7.0/blob/9231372c049a3f4df4fc47d5e5b78c55c179ce27/web/concrete/src/Url/Url.php#L9) that gets called with `$url = ''` and `$trailing_slashes = false`, and there I can't see anything that's affected by this pull request, and `$components['port']` is null...

EC-Joe commented:
Not working for me either

KorvinSzanto commented:
You're missing `\Concrete\Core\Url\Resolver\PathUrlResolver::handleHost`, that's why it's not working for these guys.

KorvinSzanto commented:
I think we'll have to add a new configuration value to this since host doesn't include port. Other frameworks have done something like:
```php
<?php
list($host, $port) = explode(':', $host_config, 2);
```
but I don't like that idea. I propose `concrete.seo.canonical_port`

EC-Joe commented:
@KorvinSzanto can you explain the benefit of canonical_port as a configuration value as opposed to simply using whatever one the user attempts to access the site on? This feels like one more thing to touch as I move around my development environments and I'm not seeing the benefit of it being configurable.

KorvinSzanto commented:
@EC-Joe because you might not access the site on the canonical url or the canonical port.

KorvinSzanto commented:
I keep making these partial replies, I'm an idiot.

@EC-Joe also this only overrides the request value in `\Concrete\Core\Url\Resolver\PathUrlResolver::handleHost` if set, it it's not set it would default to the request value.

aembler commented:
@KorvinSzanto Adding canonical port setting sounds good, and adding that setting to the URLs page. Certainly not required by default, and not intrusive, since 99.999% of people won't ever need to set this parameter to anything, or they shouldn't have to.

Mnkras commented:
Ok, I actually fixed it, I had modified that file Korvin mentioned but I didn't commit it originally, I also added the new configuration option for setting the port.

aembler commented:
This is great, thanks Mnkras
